src/dir.cr

```diff
@@ -93,6 +96,24 @@ class Dir
     String.new(dir).tap { C.free(dir as Void*) }
   end
 
+  def self.chdir path
+    if C.chdir(path) != 0
+      raise Errno.new("Error while changing directory")
+    end
+  end
+
+  def self.chdir(path)
+    old = working_directory
+    chdir(path)
+    yield
```

Comment:
Whether a block is expected, either implicitly via `yield` or explicitly via a `&block` parameter, or not counts into the method signature, thus this is actually an overload.